### PR TITLE
Modularize ecdh and publicencrypt

### PR DESCRIPTION
**Pull request by calvinmetcalf**
pulls out and modularizes ecdh and publicEncrypt, also updates and improves the tests for each and lastly adds node 0.12 and iojs to the travis matrix to make sure they pass.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,14 +43,23 @@ var dh = require('diffie-hellman');
...
-require('create-ecdh/inject')(module.exports, exports);
-require('public-encrypt/inject')(module.exports, exports);
 
-// the least I can do is make error messages for the rest of the node.js/crypto api.
+exports.createECDH = require('create-ecdh')
+
+var publicEncrypt = require('public-encrypt');
+
 [
-  'createCredentials',
```
Comment (dcousens) [inline index.js]: what happened to `createCredentials`?

**Comment by dcousens** [comment]
ACK

**Comment by calvinmetcalf** [comment]
v3.9.13

**Comment by calvinmetcalf** [comment]
I think sign is the only one left

**Pull request merged**